### ISSUE TITLE
dnsdist: LogAction: log protocol (TCP/UDP)

### ISSUE THREAD

**Issue by 0x42h** [PowerDNS/pdns#12376]
- Program: dnsdist
 - Issue type: Feature request

We're are dealing with DDoS-attacks and would like to get insight in what kind of packets are coming from where and it's a thorn in my eye that we cannot tell the difference from TCP and UDP from the LogAction-function in our dnsdist.conf/LUA script.

Could that be added?

**Comment by phonedph1** [comment]
Depending on how you log you can `AndRule` this with a `TCPRule` as a stop gap.

**Comment by 0x42h** [comment]
Yeah, I figured I could write two rules to log to two different log files with that; one for UDP and one for TCP and then use seperate `LogAction`s on those, but I thought that in one log file would be finer and sufficiently rudimentary to be included in that generic log line. Or do you mean something else?

Would, constructing something myself, with something like `warnlog` get me into trouble with high volume requests (~100k)? I'm not going to write them all to disk, of course. Just a fraction of that (`ProbaRule(0.005)`).

**Comment by rgacogne** [comment]
It would be very easy to implement that, sure, and I would gladly merge such a pull request :-)

**Comment by 0x42h** [comment]
Please hang on, @rgacogne, may somebody not go before, though, of course. Bonne journée! 😉
